### ISSUE TITLE
Packaging doesn't include docx type files.

### ISSUE THREAD

**Issue by musyilmaz** [chentsulin/electron-react-boilerplate#1506]
I am using a docxtemplater on main.dev which is called with ipc.
In here, I write data to different docx templates which lies in /app/main-helpers folder.

The sample code that I use for the templatePath can be find in below.

`
const templatePath = path.join(
      path.dirname(__filename),
      'main-helpers',
      'docx-templates',
      'single-location-count-control-template.docx'
    );
`

Before packaging everything works due to webpack is working in /app folder and the ipc event can find the docx items. After packaging the ipc event cannot find the path, probably because it is not packaged.

Any idea on how to load this folder with webpack on packaging.

**Comment by dayfine** [comment]
#1502 See my response here. Basically you need put path to your template folder [here](https://github.com/chentsulin/electron-react-boilerplate/blob/586b84f0ed5d1ded13ab296ba1219d26c70bb9dc/package.json#L37)

**Comment by musyilmaz** [comment]
Actually this addition, didnt worked for me. I am suspecting that both my main.dev.js and main-helpers/docx-templates/{actual_template.docx} is inside the app folder. 

So when I implement a build files structure like your offering, It still not working. 

`"files": [
      "dist/",
      "node_modules/",
      "app.html",
      "app/main-helpers",
      "main.prod.js",
      "main.prod.js.map",
      "package.json"
    ],`

**Comment by musyilmaz** [comment]
Is there a way that, I can unpack the asar and see how the files are implemented so I can adjust the static path formation based on this?

**Comment by dayfine** [comment]
@musyilmaz hmmm, right above the "files" line, i.e. in the `build` object, add a key-value pair of `"asar": false`

**Comment by musyilmaz** [comment]
It seems that the files still lies in in the same folder level so when I added as main-helpers/ it worked.

**Comment by musyilmaz** [comment]
Based on @dayfine 's comments how to solve this issue.

I made a deep analysis of our code that is using this boilerplate as its base. The problem is occurring because of the reasons that dayfine is indicated.

Inside the main package.json, we have a build configuration that shows the files as an Array. This array is implemented to the project as the bare minimum. Thanks to webpack, these things will be packaged in most of the projects. In our case, the problem was we were never importing the mentioned folder, so webpack cannot handle to carry them to either main.prod.js or dist/ folder.

With investigating the asar resources structure, I have understanded that the main.prod.js is still lying in the same level as in development environment app/.

So adding the folders that is never imported to any of the project files (only defined as absolute paths) is helped me solve this bug (along with 2 unpackaged folders).

As it can be seen that main.prod.js is directly listed in the files section, you should import all other folder and files based on main.dev.js's folder level.

**Comment by jakinov** [comment]
@musyilmaz Are your docx files embeded in your exe/dmg when you did this? If so, how did you reference the files embeded in your binary instead of the ones on the user's filesystem?

**Comment by musyilmaz** [comment]
Sorry for really late response. 

@jakinov yes currently we have our template files embeded in our exe/dmg and their versions are controlled by a basic yml check on application load with using an updated version system on our company dropbox. 

referencing the files is in the packaged application is using their paths but for building/embedding them in the application packaging process, I have added a single line to main package.json file in the application. I should mention that we are using a freezed repo version for our application so this might not work with the latest version.

in the main package.json there is an area that defines the build parameters as an object. in this build parameters section there is an important list that is used to package in the exe/dmg. 
This is our current build.files parameter

```
files": [
      "dist/",
      "node_modules/",
      "app.html",
      "main-helpers/",
      "main.prod.js",
      "main.prod.js.map",
      "package.json"
    ],
```

**Comment by Ancinek** [comment]
@musyilmaz Hey there, I'm having a similar issues. In the `app` folder I have defined a `scripts` directory with couple of binary scripts inside. I've added this do the `files` so it currently looks like this:

```
"files": [
      "dist/",
      "node_modules/",
      "scripts/",
      "app.html",
      "main.prod.js",
      "main.prod.js.map",
      "package.json"
    ],
```

When I run `npm run package` the files are still not included in the `.asar` package :/ Do I have to import those files from the code (so they go through webpack)? Would really appreciate your help!
